notebooks/viz/Plotly Maps.snb

```diff
@@ -0,0 +1,196 @@
+      },
+      "id" : "00119B23DAE34BD28E62C715A2BE6DEB"
+    },
+    "cell_type" : "code",
+    "source" : "val scale = 50000\nval populationData = scala.io.Source.fromURL(\"https://raw.githubusercontent.com/plotly/datasets/master/2014_us_cities.csv\")\n  .mkString.split(\"\\n\").drop(1)\n  .map{line => \n       val Array(city, pop, lat, lon) = line.split(\",\")\n       (city.trim, pop.trim.toInt, lat.trim.toDouble, lon.trim.toDouble)\n      }.toSeq\n  .toDF(\"city\", \"pop\", \"cityLat\", \"cityLon\")\n  .withColumn(\"citySize\", $\"pop\" / scale)\n  .withColumn(\"hoverText\", concat($\"city\", lit(\" pop: \"), $\"pop\"))\n  .drop(\"city\").drop(\"pop\")",
+    "outputs" : [ ]
+  }, {
+    "metadata" : {
+      "trusted" : true,
+      "input_collapsed" : false,
+      "collapsed" : false,
+      "id" : "69443643D0E64D34B2A7F129EC835548"
+    },
+    "cell_type" : "code",
+    "source" : "CustomPlotlyChart(\n  populationData, \n  layout=\"\"\"{\n        title: '2014 US City Populations',\n        showlegend: false,\n        geo: {\n            scope: 'usa',\n            projection: {\n                type: 'albers usa'\n            },\n            showland: true,\n            landcolor: 'rgb(217, 217, 217)',\n            subunitwidth: 1,\n            countrywidth: 1,\n            subunitcolor: 'rgb(255,255,255)',\n            countrycolor: 'rgb(255,255,255)'\n        },\n    }\"\"\",\n  dataOptions=\"\"\"{\n    type: 'scattergeo',\n    locationmode: 'USA-states',\n    hoverinfo: 'text',\n    marker: {\n        line: {\n            color: 'black',\n            width: 2\n        }}}\"\"\",\n  dataSources=\"{lat: 'cityLat', lon: 'cityLon', text: 'hoverText', marker: {size: 'citySize'}}\",\n  maxPoints=3500)",
+    "outputs" : [ ]
+  }, {
+    "metadata" : {
+      "id" : "4D8308053BF444B386A63B7D940ADB04"
+    },
+    "cell_type" : "markdown",
+    "source" : "# Choropleth Maps"
+  }, {
+    "metadata" : {
+      "id" : "B4A41A716E3F433898AC89C96A37595C"
+    },
+    "cell_type" : "markdown",
+    "source" : "## World Choropleth Map (Robinson Projection)"
+  }, {
+    "metadata" : {
+      "trusted" : true,
+      "input_collapsed" : false,
+      "collapsed" : false,
+      "id" : "E434B016D6534101A934A815873025AE"
+    },
+    "cell_type" : "code",
+    "source" : "val alcoholConsumption = scala.io.Source.fromURL(\"https://raw.githubusercontent.com/plotly/datasets/master/2010_alcohol_consumption_by_country.csv\")\n  .mkString.split(\"\\n\").drop(1)\n  .map{line => \n       val Array(location, alcohol) = line.split(\",\")\n       (location.trim, alcohol.trim.toDouble)\n      }.toSeq\n  .toDF(\"location\", \"alcohol\")",
```

Comment:
dumb idea, I wonder can't spark load json dataset/dataframe from https directly? would be one liner... ;)

or maybe even downloading it to local file and loading with `spark-csv` maybe would be a little cleaner too ?